aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java

```diff
@@ -50,6 +52,19 @@
    * @param location fully qualified URI
    */
   S3URI(String location) {
+    this(location, ImmutableMap.of());
+  }
+
+  /**
+   * Creates a new S3URI in the form of scheme://bucket/key?query#fragment
+   * <p>
+   * The URI supports any valid URI schemes to be backwards compatible with s3a and s3n,
+   * and also allows users to use S3FileIO with other S3-compatible object storage services like GCS.
+   *
+   * @param location fully qualified URI
+   * @param bucketToAccessPointMapping contains mapping of bucket to access point
+   */
+  S3URI(String location, Map<String, String> bucketToAccessPointMapping) {
     Preconditions.checkNotNull(location, "Location cannot be null.");
```

Comment:
Nit: I'm not sure if `propertiesWithPrefix()` already ensures this but could you add a null check for the `bucketToAccessPointMapping` as well here. This cannot be null.

Reply:
As per my understanding it's ensured it will return empty map rather than null in this scenario, (tested it locally as well). 
https://github.com/apache/iceberg/blob/e9a287b7f68d8e7138c47dcb686d3709bdde402e/core/src/main/java/org/apache/iceberg/util/PropertyUtil.java#L85-L97


I agree the on the adding the check, as this being a public method, and can be accessed in future with different code-paths, adding this check.